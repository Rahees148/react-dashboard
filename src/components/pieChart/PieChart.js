import React, { useEffect, useState } from 'react';

import { Style } from './PieChart.style';

//Context
import {useContext} from 'react';
import {TaskContext} from '../../context/TaskContext';

function PieChart(){
    const context = useContext(TaskContext);
    const [totalCompletedTask, setCompletedTaskTotal] = useState(0);
    const tasks = context.tasks;
    useEffect(()=>{
        const completedTask = tasks.filter(task => task.completed === true);
        const completedTasksPercentage = (completedTask.length/tasks.length) * 100
        setCompletedTaskTotal(completedTasksPercentage);
    },[tasks]);

    return (
        <Style.Card>
          <Style.Pie size={totalCompletedTask} >
            {totalCompletedTask !== 0 && <Style.Text>Completed Tasks</Style.Text> }
          </Style.Pie>
        </Style.Card>
    )
}

export default PieChart;