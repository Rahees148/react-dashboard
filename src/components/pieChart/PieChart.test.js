
import renderer from 'react-test-renderer';
import PieChart from './PieChart';


it('renders correctly', () => {
  const tree = renderer
    .create(<PieChart />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});