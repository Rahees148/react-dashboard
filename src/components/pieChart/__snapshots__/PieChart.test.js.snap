// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdfBwQ gAQogk"
>
  <div
    className="sc-gsTCUz YJpp"
    size={0}
  />
</div>
`;
